src/components/MissingDataPopup/MissingDataPopup.tsx

```diff
@@ -64,7 +73,20 @@ const PopupContent: FC<PopupContentProps> = ({ values, setOpen }) => {
       <div className={styles.footer}>
         <ActionButton size="small" type="decline" onClick={(): void => setOpen(false)} />
         {showCheck && (
-          <ActionButton size="small" type="accept" onClick={(): void => setOpen(false)} />
+          <ActionButton
+            size="small"
+            type="accept"
+            onClick={(): void => {
+              setOpen(false);
+              // Create DocumentReference with selected value and add to patient record
+              if (note) {
+                const noteString = createNoteContent(note, patientRecords, 'completed', selected);
+                const documentReference = createDocumentReference(noteString, selected, patient);
+                setPatientRecords([...patientRecords, documentReference]);
+                client?.create?.(documentReference);
```

Comment:
You can actually see it on a DSTU2 server -- for some reason it fails because there's an `id` included in the resource, even though it works fine on an R4 server. Yes we still see the DocumentReference in the patientrecords, even though it's not on the server. I don't want to hold this up for error handling though, let's make a backlog item for that.